dev/PushNotifications/PushNotificationReceivedEventArgs.cpp

```diff
@@ -60,54 +49,30 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
         return { payload, payload + (length * sizeof(uint8_t)) };
     }
 
-    std::vector<uint8_t> PushNotificationReceivedEventArgs::BuildPayload(winrt::hstring const& payload)
-    {
-        std::string payloadToSimpleString{ ::winrt::Microsoft::Windows::PushNotifications::Helpers::WideStringToUtf8String(payload) };
-        return { payloadToSimpleString.c_str(), payloadToSimpleString.c_str() + (payloadToSimpleString.length() * sizeof(uint8_t)) };
-    }
-
     winrt::com_array<uint8_t> PushNotificationReceivedEventArgs::Payload()
     {
         return { m_rawNotificationPayload.data(), m_rawNotificationPayload.data() + (m_rawNotificationPayload.size() * sizeof(uint8_t)) };
     }
 
     winrt::BackgroundTaskDeferral PushNotificationReceivedEventArgs::GetDeferral()
     {
-        if (!m_unpackagedAppScenario)
-        {
-            THROW_HR_IF_NULL_MSG(E_ILLEGAL_METHOD_CALL, m_backgroundTaskInstance, "Foreground activation cannot call this.");
+        THROW_HR_IF_MSG(E_ILLEGAL_METHOD_CALL, m_isForegroundActivation, "Foreground activation cannot call this.");
```

Comment:
So what happened in this PR is that instead of switching on unpackaged | packagedAppScenario, we switch on is it foreground | background, which makes more sense because PushActivation contains both foreground/background. 

The reason we no longer have to switch on unpackaged/packaged is because we are correctly mocking the IBackgroundTaskInstance and unpackaged scenarios can call into the background functions and not throw (albeit they will receive dummy values).

Reply:
DCRs are only for late features and this is really just a refactoring task, so we can wait for the 1.3 payload to ship this.

Reply:
I see what you are saying about the behavior change. 

IMO, it feels like it is a product code bug to be able to return background features of PushActivatedEventArgs in a foreground activation context without throwing. This PR correctly throws if the app was activated via the foreground and the app requests a background feature.

Reply:
What do you think about this change? Should it go through review before moving to product code? @DrusTheAxe

Reply:
Reverted change